File: common-core/src/main/java/com/ityu/bean/vo/front/Rets.java
package com.ityu.bean.vo.front;



public class Rets {

    public static final Integer SUCCESS = 0;
    public static final Integer FAILURE = 1;
    public static final Integer TOKEN_EXPIRE = 50014;

    public static Ret success(Object data) {
        return new Ret(Rets.SUCCESS, "成功", data);
    }

    public static Ret error(Integer code, String msg) {
        return new Ret(code, msg, null);
    }

    public static Ret failure(String msg) {
        return new Ret(Rets.FAILURE, msg, null);
    }

    public static Ret success() {
        return new Ret(Rets.SUCCESS, "成功", null);
    }

    public static Ret expire() {
        return new Ret(Rets.TOKEN_EXPIRE, "token 过期", null);
    }
}

File: common-core/src/main/java/com/ityu/common/bean/vo/front/Rets.kt
package com.ityu.common.bean.vo.front

import com.ityu.common.exception.IErrorCode


object Rets {
    const val SUCCESS = 20000
    const val FAILURE = 9999
    const val TOKEN_EXPIRE = 50014
    fun success(data: Any?): Ret<*> {
        return Ret(SUCCESS, "成功", data)
    }

    fun failure(msg: String = "", error: IErrorCode? = null): Ret<*> {
        return when (error) {
            null -> Ret<Any?>(FAILURE, msg, null)
            else -> Ret<Any?>(error.code, error.message, null)
        }
    }

    fun success(): Ret<*> {
        return Ret<Any?>(SUCCESS, "成功", null)
    }

    fun expire(): Ret<*> {
        return Ret<Any?>(TOKEN_EXPIRE, "token 过期", null)
    }
}
